File: packages/lib-backend-js/src/server/utils/Server/_Server.models.ts
import { type ApiConfigModel, type ApiEndpointModel } from '@lib/config/api/api.models';
import { type BootstrappableModel } from '@lib/shared/core/utils/Bootstrappable/Bootstrappable.models';
import { type FastifyInstance } from 'fastify';

export type _ServerParamsModel = {
  api?: ApiConfigModel;

  certificate?: {
    caFilename: string;

    certificateDir: string;

    privateKeyFilename: string;

    publicKeyFilename: string;
  };

  host: string;

  port?: number;
};

export type _ServerModel = BootstrappableModel & {
  _app: FastifyInstance;

  register<TType, TParams>(endpoint: ApiEndpointModel<TType, TParams>): Promise<void>;

  run(): Promise<void>;
};

File: packages/lib-config-py/src/lib_config/http/server/server_models.py
from typing import Optional

from lib_shared.core.utils.base_model import BaseModel

from lib_config.http.api.api_models import ApiConfigModel


class ServerConfigModel(BaseModel):
    api: ApiConfigModel
    host: Optional[str] = "127.0.0.1"
    port: Optional[int | str] = 5000
